plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -56,9 +56,10 @@ def update_replica_count(new_count)
 
       if replicas_live >= replica_target
         self.status ='live'
-      elsif replicas_live > 0
+      elsif replicas_live > 0 && !failed?
```

Comment:
There might be a glitch when deploying after which resources for scheduling a pod become available, I've seen this in ad-hoc tests. We kill the deploy watcher only after the deploy is live or there was a timeout, to give it a chance of recovery. The UI will flicker but at least we won't cause panic over a good deploy.